src/provider/apiresponse.h

```diff
@@ -0,0 +1,54 @@
+#ifndef ERD_PROXY_API_RESPONSE_H
+#define ERD_PROXY_API_RESPONSE_H
+
+#include "json/json.hpp"
+#include "../utils/errors.h"
+
+class ErdGenericApiResponse
+{
+public:
+    explicit ErdGenericApiResponse(std::string rawData)
+    {
+        try
+        {
+            m_response = nlohmann::json::parse(rawData);
+        }
+        catch (...)
+        {
+            throw std::invalid_argument(ERROR_MSG_JSON_SERIALIZED + rawData);
+        }
+    }
+
+    std::string getError() const
+    {
+        std::string ret;
+
+        if (m_response.contains("error"))
+            if(m_response["error"] != "")
+                ret = m_response["error"];
```

Comment:
Just return, don't define & assign? Fine as it is, though.

About code formatting - aren't the curly braces mandatory in our set of rules (here, and below, in other functions)?

Reply:
> Just return, don't define & assign?

I've noticed that for different API urls, we get different responses.
E.g.: 
- When sending a valid transaction with URL = https://testnet-api.elrond.com , we **don't** receive **error** attribute,
- When sending a valid transaction with URL = https://testnet-gateway.elrond.com , we receive _**empty**_ **error** attribute.

This is the reason why the code looks like that.

> About code formatting - aren't the curly braces mandatory in our set of rules (here, and below, in other functions)?

I guess you are right, I've added them !

Reply:
They are different APIs. For example, in erdjs, there are two separate "providers" (API client wrappers): one for the General API, one for the Gateway (Proxy API). Perhaps we should have two separate providers in erdcpp as well, at some point?